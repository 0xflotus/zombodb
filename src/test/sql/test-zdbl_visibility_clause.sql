select jsonb_pretty(regexp_replace(zdb.visibility_clause(ARRAY[42]::bigint[], 99, 0, ARRAY[1,2,3,4,5]::bigint[], 'idxevents', 'doc')::text, '"index":".*?"', '"index":"xxx"', 'g')::jsonb);