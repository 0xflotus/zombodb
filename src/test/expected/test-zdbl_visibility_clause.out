select jsonb_pretty(regexp_replace(zdb.visibility_clause(ARRAY[42]::bigint[], 99, 0, ARRAY[1,2,3,4,5]::bigint[], 'idxevents', 'doc')::text, '"index":".*?"', '"index":"xxx"', 'g')::jsonb);
                                                        jsonb_pretty                                                         
-----------------------------------------------------------------------------------------------------------------------------
 {                                                                                                                          +
     "bool": {                                                                                                              +
         "must": [                                                                                                          +
             {                                                                                                              +
                 "bool": {                                                                                                  +
                     "must_not": [                                                                                          +
                         {                                                                                                  +
                             "query_string": {                                                                              +
                                 "query": "_id:zdb_aborted_xids"                                                            +
                             }                                                                                              +
                         }                                                                                                  +
                     ]                                                                                                      +
                 }                                                                                                          +
             },                                                                                                             +
             {                                                                                                              +
                 "bool": {                                                                                                  +
                     "should": [                                                                                            +
                         {                                                                                                  +
                             "bool": {                                                                                      +
                                 "must": [                                                                                  +
                                     {                                                                                      +
                                         "terms": {                                                                         +
                                             "zdb_xmin": [                                                                  +
                                                 42                                                                         +
                                             ]                                                                              +
                                         }                                                                                  +
                                     },                                                                                     +
                                     {                                                                                      +
                                         "range": {                                                                         +
                                             "zdb_cmin": {                                                                  +
                                                 "lt": "0"                                                                  +
                                             }                                                                              +
                                         }                                                                                  +
                                     },                                                                                     +
                                     {                                                                                      +
                                         "bool": {                                                                          +
                                             "should": [                                                                    +
                                                 {                                                                          +
                                                     "bool": {                                                              +
                                                         "must_not": [                                                      +
                                                             {                                                              +
                                                                 "exists": {                                                +
                                                                     "field": "zdb_xmax"                                    +
                                                                 }                                                          +
                                                             }                                                              +
                                                         ]                                                                  +
                                                     }                                                                      +
                                                 },                                                                         +
                                                 {                                                                          +
                                                     "bool": {                                                              +
                                                         "must": [                                                          +
                                                             {                                                              +
                                                                 "terms": {                                                 +
                                                                     "zdb_xmax": [                                          +
                                                                         42                                                 +
                                                                     ]                                                      +
                                                                 }                                                          +
                                                             },                                                             +
                                                             {                                                              +
                                                                 "range": {                                                 +
                                                                     "zdb_cmax": {                                          +
                                                                         "gte": "0"                                         +
                                                                     }                                                      +
                                                                 }                                                          +
                                                             }                                                              +
                                                         ]                                                                  +
                                                     }                                                                      +
                                                 }                                                                          +
                                             ]                                                                              +
                                         }                                                                                  +
                                     }                                                                                      +
                                 ]                                                                                          +
                             }                                                                                              +
                         },                                                                                                 +
                         {                                                                                                  +
                             "bool": {                                                                                      +
                                 "must": [                                                                                  +
                                     {                                                                                      +
                                         "bool": {                                                                          +
                                             "must": [                                                                      +
                                                 {                                                                          +
                                                     "bool": {                                                              +
                                                         "must_not": [                                                      +
                                                             {                                                              +
                                                                 "terms": {                                                 +
                                                                     "zdb_xmin": {                                          +
                                                                         "id": "zdb_aborted_xids",                          +
                                                                         "path": "zdb_aborted_xids",                        +
                                                                         "type": "doc",                                     +
                                                                         "index": "xxx"                                     +
                                                                     }                                                      +
                                                                 }                                                          +
                                                             }                                                              +
                                                         ]                                                                  +
                                                     }                                                                      +
                                                 },                                                                         +
                                                 {                                                                          +
                                                     "bool": {                                                              +
                                                         "must_not": [                                                      +
                                                             {                                                              +
                                                                 "terms": {                                                 +
                                                                     "zdb_xmin": [                                          +
                                                                         1,                                                 +
                                                                         2,                                                 +
                                                                         3,                                                 +
                                                                         4,                                                 +
                                                                         5                                                  +
                                                                     ]                                                      +
                                                                 }                                                          +
                                                             }                                                              +
                                                         ]                                                                  +
                                                     }                                                                      +
                                                 },                                                                         +
                                                 {                                                                          +
                                                     "bool": {                                                              +
                                                         "must_not": [                                                      +
                                                             {                                                              +
                                                                 "range": {                                                 +
                                                                     "zdb_xmin": {                                          +
                                                                         "gte": "99"                                        +
                                                                     }                                                      +
                                                                 }                                                          +
                                                             }                                                              +
                                                         ]                                                                  +
                                                     }                                                                      +
                                                 },                                                                         +
                                                 {                                                                          +
                                                     "bool": {                                                              +
                                                         "should": [                                                        +
                                                             {                                                              +
                                                                 "bool": {                                                  +
                                                                     "must_not": [                                          +
                                                                         {                                                  +
                                                                             "exists": {                                    +
                                                                                 "field": "zdb_xmax"                        +
                                                                             }                                              +
                                                                         }                                                  +
                                                                     ]                                                      +
                                                                 }                                                          +
                                                             },                                                             +
                                                             {                                                              +
                                                                 "bool": {                                                  +
                                                                     "must": [                                              +
                                                                         {                                                  +
                                                                             "terms": {                                     +
                                                                                 "zdb_xmax": [                              +
                                                                                     42                                     +
                                                                                 ]                                          +
                                                                             }                                              +
                                                                         },                                                 +
                                                                         {                                                  +
                                                                             "range": {                                     +
                                                                                 "zdb_cmax": {                              +
                                                                                     "gte": "0"                             +
                                                                                 }                                          +
                                                                             }                                              +
                                                                         }                                                  +
                                                                     ]                                                      +
                                                                 }                                                          +
                                                             },                                                             +
                                                             {                                                              +
                                                                 "bool": {                                                  +
                                                                     "must": [                                              +
                                                                         {                                                  +
                                                                             "bool": {                                      +
                                                                                 "must_not": [                              +
                                                                                     {                                      +
                                                                                         "terms": {                         +
                                                                                             "zdb_xmax": [                  +
                                                                                                 42                         +
                                                                                             ]                              +
                                                                                         }                                  +
                                                                                     }                                      +
                                                                                 ]                                          +
                                                                             }                                              +
                                                                         },                                                 +
                                                                         {                                                  +
                                                                             "bool": {                                      +
                                                                                 "should": [                                +
                                                                                     {                                      +
                                                                                         "terms": {                         +
                                                                                             "zdb_xmax": {                  +
                                                                                                 "id": "zdb_aborted_xids",  +
                                                                                                 "path": "zdb_aborted_xids",+
                                                                                                 "type": "doc",             +
                                                                                                 "index": "xxx"             +
                                                                                             }                              +
                                                                                         }                                  +
                                                                                     },                                     +
                                                                                     {                                      +
                                                                                         "terms": {                         +
                                                                                             "zdb_xmax": [                  +
                                                                                                 1,                         +
                                                                                                 2,                         +
                                                                                                 3,                         +
                                                                                                 4,                         +
                                                                                                 5                          +
                                                                                             ]                              +
                                                                                         }                                  +
                                                                                     },                                     +
                                                                                     {                                      +
                                                                                         "range": {                         +
                                                                                             "zdb_xmax": {                  +
                                                                                                 "gte": "99"                +
                                                                                             }                              +
                                                                                         }                                  +
                                                                                     }                                      +
                                                                                 ]                                          +
                                                                             }                                              +
                                                                         }                                                  +
                                                                     ]                                                      +
                                                                 }                                                          +
                                                             }                                                              +
                                                         ]                                                                  +
                                                     }                                                                      +
                                                 }                                                                          +
                                             ]                                                                              +
                                         }                                                                                  +
                                     }                                                                                      +
                                 ]                                                                                          +
                             }                                                                                              +
                         }                                                                                                  +
                     ]                                                                                                      +
                 }                                                                                                          +
             }                                                                                                              +
         ]                                                                                                                  +
     }                                                                                                                      +
 }
(1 row)

